tests/test_mean_estimator.py

```diff
@@ -1,733 +1,166 @@
-                    9.82006976e-02,
-                ],
-                atol=1e-4,
-            )
+@pytest.fixture(scope="module")
+def mask(L):
+    return grid_3d(L)["r"] < 1
+
+
+# Tests
+def test_resolution_error(sim, basis):
+    """
+    Test mismatched resolutions yields a relevant error message.
+    """
+
+    with raises(ValueError, match=r".*resolution.*"):
+        # This basis is intentionally the wrong resolution.
+        incorrect_basis = FBBasis3D(sim.L + 1, dtype=sim.dtype)
+
+        _ = MeanEstimator(sim, basis=incorrect_basis, preconditioner="none")
+
+
+def test_estimate(sim, estimator, mask):
+    estimate = estimator.estimate()
+
+    est = estimate * mask
+    vol = sim.vols * mask
+
+    np.testing.assert_allclose(
+        est / np.linalg.norm(est), vol / np.linalg.norm(vol), atol=0.1
+    )
+
+
+def test_adjoint(sim, basis, estimator, mask):
+    # Mean coefs formed by backprojections
+    mean_b_coef = estimator.src_backward()
+
+    # Evaluate mean coefs into a volume
+    est = Coef(basis, mean_b_coef).evaluate() * mask
+
+    # Mask off corners of volume
+    vol = sim.vols * mask
+
+    # Assert the mean volume is close to original volume
+    np.testing.assert_allclose(
+        est / np.linalg.norm(est), vol / np.linalg.norm(vol), atol=0.11
+    )
+
+
+def test_checkpoint(sim, basis, estimator):
+    """Exercise the checkpointing and max iterations branches."""
+    test_iter = 2
+    with tempfile.TemporaryDirectory() as tmp_input_dir:
+        prefix = os.path.join(tmp_input_dir, "new", "dirs", "chk")
+        _estimator = MeanEstimator(
+            sim,
+            basis=basis,
+            preconditioner="none",
+            checkpoint_iterations=test_iter,
+            maxiter=test_iter + 1,
+            checkpoint_prefix=prefix,
         )
 
-    def testCheckpoint(self):
-        """Exercise the checkpointing and max iterations branches."""
-        test_iter = 2
-        with tempfile.TemporaryDirectory() as tmp_input_dir:
-            prefix = os.path.join(tmp_input_dir, "new", "dirs", "chk")
-            estimator = MeanEstimator(
-                self.sim,
-                basis=self.basis,
-                preconditioner="none",
-                checkpoint_iterations=test_iter,
-                maxiter=test_iter + 1,
-                checkpoint_prefix=prefix,
-            )
-
-            # Assert we raise when reading `maxiter`.
-            with raises(RuntimeError, match="Unable to converge!"):
-                _ = estimator.estimate()
-
-            # Load the checkpoint coefficients while tmp_input_dir exists.
-            b_chk = np.load(f"{prefix}_iter{test_iter:04d}.npy")
+        # Assert we raise when reading `maxiter`.
+        with raises(RuntimeError, match="Unable to converge!"):
+            _ = _estimator.estimate()
+
+        # Load the checkpoint coefficients while tmp_input_dir exists.
+        b_chk = np.load(f"{prefix}_iter{test_iter:04d}.npy")
```

Comment:
Is it the b (RHS) coefficients that are stored or the current estimate coefficients?

Reply:
Ah, oops, poor name on my part.  The checkpoint saves are the `x` estimate from the last iteration.  I'll change to `x` something, thanks!

Reply:
Passing through to `x0`, renamed `x_chk`. My mistake.

Reply:
Thanks!